application/modules/upload/controllers/Upload.php

```diff
@@ -22,6 +22,15 @@ class Upload extends Admin_Controller
 
     public $content_types = [];
 
+    private $allowed_extensions = ['jpg', 'jpeg', 'png', 'gif', 'webp'];
```

Comment:
the `$this->mdl_uploads->content_types` model already defines the authoritative whitelist

Reply:
If $this->content_types is only used once, don't add it to constructor, but just inline:
```
$this->content_types = $this->mdl_uploads->content_types;
```